### PR TITLE
X509 resource remaining

### PR DESCRIPTION
**Pull request by juju482**
### Description

Hello, this is another contribution from the french organization [Institut National de l'Audiovisuel](https://institut.ina.fr)

it is a big remaining of the x509 resource with backward compatibility, documentation & (a lot of) tests. it adds the new functionality below 

- openssl_x509 is renamed to openssl_x509_certificate
- openssl_x509_certificate can now generate a signed certificate with a provided CA cert & key
- openssl_x509_certificate now support x509 extensions
- openssl_x509_certificate now support x509 csr
- openssl_x509_certificate now generate a random serial for the certificate

I also fix a few bugs

- expires has now a default value : 365
- country field is now mandatory in x509_request
- the private key file is not rewrited in x509_request if it already exist

I know this is a big remaining, take your time and tell me if any modification is needed.

### Issues Resolve

https://github.com/chef-cookbooks/openssl/issues/42

### Check List

- [x] All tests pass. See <https://github.com/chef-cookbooks/community_cookbook_documentation/blob/master/TESTING.MD>
- [x] New functionality includes testing.
- [x] New functionality has been documented in the README if applicable
- [x] All commits have been signed for the Developer Certificate of Origin. See <https://github.com/chef-cookbooks/community_cookbook_documentation/blob/master/CONTRIBUTING.MD>

### CODE REVIEW

**Comment by tas50** [comment]
This looks great. I'll carve out some time to check it out tomorrow

File: resources/x509_certificate.rb [hunk 1]
```diff
@@ -0,0 +1,134 @@
+provides :openssl_x509 # legacy_name
+provides :openssl_x509_cert
```
Comment (tas50) [inline resources/x509_certificate.rb]: Do we need this resource name as well?

File: resources/x509_certificate.rb [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+provides :openssl_x509_certificate
+
+include OpenSSLCookbook::Helpers
+
+property :path,             String, name_property: true
+property :owner,            String, default: node['platform'] == 'windows' ? 'Administrator' : 'root'
```
Comment (tas50) [inline resources/x509_certificate.rb]: You can lazy { node['root_group'] } here

Comment (tas50) [inline resources/x509_certificate.rb]: You can shrink this a bit with platform?('windows') ? 'Administrator' : 'root'

File: resources/x509_certificate.rb (hunk 1 above)
Comment (juju482) [reply resources/x509_certificate.rb]: openssl_x509 war for backward compatibility. I guess we should keep it. 

I was afraid openssl_x509_certificate was maybe too long so I added openssl_x509_cert... Maybe that's not  a good idea. I can remove it if needed.

File: resources/x509_certificate.rb (hunk 2 above)
Comment (juju482) [reply resources/x509_certificate.rb]: Will do it tomorrow. ;-)

Comment (tas50) [reply resources/x509_certificate.rb]: I can just do it now. Easy enough.

**Pull request merged**